python/app/dialog.py

```diff
@@ -978,13 +1022,30 @@ def build_entity_tabs(self):
             # view (QListView), filter (QCheckbox)
             if data["has_description"]:
                 label = self.create_entity_tab_label(entity_tab_name, tab_widget)
-                tab_widget.layout().addWidget(label)
+                if data["entity_type"] == "Task":
+                    # Create the Label Vertical Layout
+                    label_lay = QtGui.QHBoxLayout()
+                    label_lay.addWidget(label)
+                    # Nest the label layout to the label_menus layout
+                    self._label_menus_layout.addLayout(label_lay)
+                else:
+                    tab_widget.layout().addWidget(label)
```

Comment:
This only adds the filter/sort menu if the entity type is a Task -- this should be generic and allow adding filter/sort menu for any entity type